From 7b395d8a46e3599ebc8580e1256d7acedbafea30 Mon Sep 17 00:00:00 2001
From: Green Wan <green.wan@sifive.com>
Date: Mon, 22 Feb 2021 10:56:35 +0000
Subject: [PATCH 35/50] riscv: clear feature disable CSR

feature disable CSR should be cleared for each core to gain
performance.

Signed-off-by: Green Wan <green.wan@sifive.com>
---
 arch/riscv/cpu/start.S | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/riscv/cpu/start.S b/arch/riscv/cpu/start.S
index 8589509e01..c3d4647074 100644
--- a/arch/riscv/cpu/start.S
+++ b/arch/riscv/cpu/start.S
@@ -41,6 +41,7 @@ secondary_harts_relocation_error:
 _start:
 #if CONFIG_IS_ENABLED(RISCV_MMODE)
 	csrr	a0, CSR_MHARTID
+	csrwi	0x7c1, 0
 #endif
 
 	/*
-- 
2.31.0

